pkg/providers/qemu/stop_instance_cgo.go

```diff
@@ -23,7 +24,12 @@ func (p *QemuProvider) StopInstance(id string) error {
 		return errors.New("invalid instance id (should be qemu pid)", err)
 	}
 
-	if err := syscall.Kill(pid, syscall.SIGKILL); err != nil {
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		logrus.Warn("failed finding instance, assuming instance has externally terminated", err)
+	}
+
+	if err := process.Signal(syscall.SIGKILL); err != nil {
```

Comment:
you'll get a panic here if `process == nil`

Reply:
Thanks for pointing it out. I rechecked it. In Golang, the ``process`` will never be ``nil`` if ``err == nil``, you can check the three implementations for windows, unix and plan9 here: https://github.com/golang/go/search?q=findProcess%28pid+int%29&unscoped_q=findProcess%28pid+int%29

And Golang also officially assume it in its test:

https://github.com/golang/go/blob/516e6f6d5d83dc3dcee6403fab25d5954bbf3f62/src/net/http/cgi/posix_test.go#L14-L20

Reply:
but there's no early return here; you check `if err != nil`, log a warning and then continue (in which case `process == nil`)

Reply:
I have fixed it in the new commit. Please review.